spec/features/admin/admins/admin_create_admin_spec.rb

```diff
@@ -0,0 +1,36 @@
+require 'rails_helper'
+
+feature 'Admin can create a new User' do
+
+  signed_in_as(:admin) do
+    before do
+      click_header_option("Dashboard")
+      click_sidemenu_option("Admins")
+      click_link("Add new admin")
+    end
+
+    scenario 'Admin creates user with valid data' do
+      fill_in("Email", with: "valid@example.com")
+      fill_in("Password", with: "password")
+      click_button("Create")
+
+      # Current user should be redirected to the index
+      expect(current_path).to eq(admin_admins_path)
+
+      # User should be saved
+      latest_user = User.order(:created_at).last
+      expect(latest_user.email).to eq("valid@example.com")
+      expect(latest_user).to be_admin
+    end
+
+    scenario 'Admin creates user with invalid data' do
+      fill_in("Email", with: "")
+      fill_in("Password", with: "")
+      click_button("Create")
+
+      # Ensure no user is created
+      expect(User.count).to eq(1)
+      expect(page).to have_error_message("user_email", "can't be blank")
```

Comment:
Yes, that's not something that happen often, more often when test involves javascript.

The thing is, you can prevent headaches by simply inverting the order of assertions of your feature tests.

If you see this test failing on CI what would you do?

In the best case: you would probably run the test once, twice, 3 times, see the test passing locally, go to CI, trigger a new build, wait for the whole test suite to run, see it passing on CI as well, ignore the failing build and go back to work.

Worst case: run tests locally, see tests failing, visit the page and see that it actually works, spend some minutes investigating why tests still returning red.

IMO the develop effort it requires is lowermost in considering the headaches and time it saves to you.